foxtrot-core/src/main/java/com/flipkart/foxtrot/core/common/Action.java

```diff
@@ -84,6 +87,10 @@ public String call() throws Exception {
     }
 
     public ActionResponse execute() throws QueryStoreException {
+        if (!validate()) {
```

Comment:
Let validate throw exception instead of throwing same ErrorCode for all validation failures